doc/developer/platform/ux.md

```diff
@@ -299,17 +299,63 @@ The HTTPS server exposes two APIs:
     additional control over query execution and result fetching, a la the
     [Snowflake SQL REST API].
 
-    TODO: fully describe the output format.
+    ### Output format
 
-    TODO: design support for binding parameters.
+    The output format is a JSON object with one key, `results`, whose values is
+    an array of serialized `adapter::command::SimpleResult`, representing all of
+    the statements' executions.
+
+    The serialized `adapter::command::SimpleResult` values are representing in JSON as:
+
+    `SimpleResult` value | JSON value
+    ---------------------|------------
+    `Rows` | `{"rows": <2D array of JSON-ified results>, "col_names": <array of text>}`
+    `Error` | `{"error": <Error string from execution>}`
+    `Ok` | `{ "ok": <tag>}`
+
+    ### Binding parameters
+
+    You can include parameters alongside the query by including on the query a
+    key `"parameters"` whose value is an array of objects of the form:
+
+    ```json
+    {
+      "type": "<resolvable type name>",
+      "value": "<JSON-ified value>"
+    }
+    ```
+
+    You can then reference these parameters in the query as you would with PG,
+    i.e. using 1-indexed references to the parameters array, prefaced by a `$`. e.g.
+
+    ```json
+    {
+      "sql": "INSERT INTO foo VALUES ($1);",
+      "parameters": [{"type": "int", "value": 1}]
```

Comment:
We might need to specify here and enforce in the code that numeric values greater than javascript's `Number.MAX_SAFE_INTEGER` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) are explicitly disallowed to be passed as numbers and must be instead passed as strings. I think this is needed to prevent silently mutated data. That kind of API miiight be really annoying because sometimes it will start complaining at you depending on the passed value. We could also decide that "all non-float numbers must be passed as strings", or just straight up "all numbers must be passed as strings". That would handle NaN and Infinity, which are problematic also in javascript: `JSON.stringify(NaN) === JSON.stringify(Number.POSITIVE_INFINITY)` is `true` for example (both are `null`).

For example, I was viewing traces in jaeger a few days ago and noticed that some of the stash collection IDs (INT8s in cockroach that start out at the extreme range) were being printed with the last 4 digits or so rounded, because jaeger doesn't correctly handle the full int64 range and should be passing those as strings instead.

Reply:
Strong preference for passing all numbers as strings!

Reply:
All strings it is.